pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go

```diff
@@ -0,0 +1,185 @@
+	}
+
+	labelMap, err := metav1.LabelSelectorAsMap(&clusterExternalSecret.Spec.NamespaceSelector)
+	if err != nil {
+		log.Error(err, errLabelMap)
+		return ctrl.Result{RequeueAfter: refreshInt}, nil
+	}
+
+	namespaceList, err := namespaces.List(ctx, metav1.ListOptions{LabelSelector: labels.SelectorFromSet(labelMap).String()})
+	if err != nil {
+		log.Error(err, errNamespaces)
+		return ctrl.Result{RequeueAfter: refreshInt}, nil
+	}
+
+	esName := clusterExternalSecret.Spec.ExternalSecretName
+	if esName == "" {
+		esName = clusterExternalSecret.ObjectMeta.Name
+	}
+
+	failedNamespaces := []string{}
+
+	for _, namespace := range namespaceList.Items {
+		var existingES esv1alpha1.ExternalSecret
+		err = r.Get(ctx, types.NamespacedName{
+			Name:      esName,
+			Namespace: namespace.Name,
+		}, &existingES)
+
+		if err != nil && !apierrors.IsNotFound(err) {
+			log.Error(err, errGetExistingES)
+		}
+
+		// No one owns this resource so error out
+		if len(existingES.ObjectMeta.OwnerReferences) == 0 {
+			log.Error(nil, errSecretAlreadyExists, "namespace", namespace)
+			failedNamespaces = append(failedNamespaces, namespace.Name)
+			continue
+		}
+
+		// this means the existing ES does not belong to us
+		if err = controllerutil.SetControllerReference(&clusterExternalSecret, &existingES, r.Scheme); err != nil {
+			log.Error(err, errSetCtrlReference, "namespace", namespace)
+			failedNamespaces = append(failedNamespaces, namespace.Name)
+			continue
+		}
+
+		externalSecret := esv1alpha1.ExternalSecret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      esName,
+				Namespace: namespace.Name,
+			},
+			Spec: clusterExternalSecret.Spec.ExternalSecretSpec,
+		}
+
+		err = controllerutil.SetControllerReference(&clusterExternalSecret, &externalSecret, r.Scheme)
+		if err != nil {
+			log.Error(err, errSetCtrlReference)
+			failedNamespaces = append(failedNamespaces, namespace.Name)
+			continue
+		}
+
+		mutateFunc := func() error {
+			externalSecret.Spec = clusterExternalSecret.Spec.ExternalSecretSpec
+			return nil
+		}
+
+		// An empty mutate func as nothing needs to happen currently
+		_, err = ctrl.CreateOrUpdate(ctx, r.Client, &externalSecret, mutateFunc)
+
+		if err != nil {
+			log.Error(err, errCreatingOrUpdating)
+			failedNamespaces = append(failedNamespaces, namespace.Name)
+		}
+	}
+
+	if len(failedNamespaces) > 0 {
+		var conditionType esv1alpha1.ClusterExternalSecretConditionType
+
+		if len(failedNamespaces) < len(namespaceList.Items) {
+			conditionType = esv1alpha1.ClusterExternalSecretPartiallyReady
+		} else {
+			conditionType = esv1alpha1.ClusterExternalSecretNotReady
+		}
+
+		// TODO maybe make this more descriptive
+		conditionFailed := NewClusterExternalSecretCondition(conditionType, v1.ConditionFalse, "one or more namespaces failed")
+		SetClusterExternalSecretCondition(&clusterExternalSecret, *conditionFailed)
+		clusterExternalSecret.Status.FailedNamespaces = failedNamespaces
```

Comment:
Lastly, I think we should also error out this Result, as it is sort of an error.
```suggestion
		clusterExternalSecret.Status.FailedNamespaces = failedNamespaces
		return ctrl.Result{RequeueAfter: refreshInt}, fmt.Errorf("failed to sync to the following namespaces:%v",failedNamespaces)
```